docs/migration_guides/opflow_migration.rst

```diff
@@ -0,0 +1,1156 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+.. raw:: html
+
+    <details>
+    <summary><a><b>Opflow</b></a></summary>
+
+.. code-block:: python
+
+    from qiskit_aer import Aer
+    from qiskit.circuit import QuantumCircuit, Parameter
+    from qiskit.opflow import ListOp, StateFn, CircuitSampler
+
+    x, y = Parameter("x"), Parameter("y")
+
+    circuit1 = QuantumCircuit(1)
+    circuit1.p(0.2, 0)
+    circuit2 = QuantumCircuit(1)
+    circuit2.p(x, 0)
+    circuit3 = QuantumCircuit(1)
+    circuit3.p(y, 0)
+
+    bindings = {x: -0.4, y: 0.4}
+    listop = ListOp([StateFn(circuit) for circuit in [circuit1, circuit2, circuit3]])
+
+    sampler = CircuitSampler(Aer.get_backend("aer_simulator"))
+    sampled = sampler.convert(listop, params=bindings).eval()
+    # returns list of SparseVectorStateFn
+
+.. raw:: html
+
+    </details>
+
+.. raw:: html
+
+    <details>
+    <summary><a><b>Alternative</b></a></summary>
+
+.. code-block:: python
+
+    from qiskit.circuit import QuantumCircuit, Parameter
+    from qiskit.primitives import Sampler
+
+    x, y = Parameter("x"), Parameter("y")
+
+    circuit1 = QuantumCircuit(1)
+    circuit1.p(0.2, 0)
+    circuit2 = QuantumCircuit(1)
+    circuit2.p(x, 0)
+    circuit3 = QuantumCircuit(1)
+    circuit3.p(y, 0)
+
+    circuits = [circuit1, circuit2, circuit3]
+    param_values = [[-0.4, 0.4] for _ in circuits]
+
+    sampler = Sampler()
+    sampled = sampler.run(circuits, param_values).result().quasi_dists
+
+.. raw:: html
+
+    </details>
+
+Example 2: ``CircuitSampler`` for computing Expectation Values
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. raw:: html
+
+    <details>
+    <summary><a><b>Opflow</b></a></summary>
+
+.. code-block:: python
+
+    from qiskit import QuantumCircuit
+    from qiskit.opflow import X, Z, StateFn, CircuitStateFn, CircuitSampler
+    from qiskit.providers.aer import AerSimulator
+
+    qc = QuantumCircuit(1)
+    qc.h(0)
+    state = CircuitStateFn(qc)
+    hamiltonian = X + Z
+
+    expr = StateFn(hamiltonian, is_measurement=True).compose(state)
+    backend = AerSimulator()
+    sampler = CircuitSampler(backend)
+    expectation = sampler.convert(expr)
+    expectation_value = expectation.eval().real
+
+.. raw:: html
+
+    </details>
+
+.. raw:: html
+
+    <details>
+    <summary><a><b>Alternative</b></a></summary>
+
+.. code-block:: python
+
+    from qiskit import QuantumCircuit
+    from qiskit.primitives import Estimator
+    from qiskit.quantum_info import SparsePauliOp
+
+    state = QuantumCircuit(1)
+    state.h(0)
+    hamiltonian = SparsePauliOp.from_list([('X', 1), ('Z',1)])
+
+    estimator = Estimator()
+    expectation_value = estimator.run(state, hamiltonian).result().values
+
+.. raw:: html
+
+    </details>
+
+Evolutions
+----------
+*Back to* `Index`_
+
+The :mod:`~qiskit.opflow.evolutions` sub-module was created to provide building blocks for hamiltonian simulation algorithms,
```

Comment:
I'm on a nitpicky streak here, but I think since `Hamiltonian` is a name it's usually spelt in uppercase 🙂 
```suggestion
The :mod:`~qiskit.opflow.evolutions` sub-module was created to provide building blocks for Hamiltonian simulation algorithms,
```